core/src/store/function.rs

```diff
@@ -0,0 +1,32 @@
+use crate::data::CustomFunction;
+use crate::result::{Error, Result};
+use async_trait::async_trait;
+
+#[async_trait(?Send)]
+pub trait Function {
+    async fn fetch_function(&self, _func_name: &str) -> Result<Option<&CustomFunction>> {
+        Err(Error::StorageMsg(
+            "[Storage] Function is not supported".to_owned(),
+        ))
+    }
+    async fn show_functions(&self) -> Result<Vec<String>> {
```

Comment:
this also implies fetching data from storage, as we use `fetch_schema` and `fetch_all_schemas`, it looks that we can also follow the patterns like.. `fetch_function` and `fetch_all_functions`?